domain/application/service/resource.go

```diff
@@ -83,14 +97,91 @@ type ContainerImageMetadataState interface {
 	) (application.ContainerImageMetadata, error)
 }
 
+// AddResources adds specified resources to an application. It validates the
+// application ID and each resource's origin. Returns a result containing
+// added resources' UUIDs or an error if validation fails.
+//
+// The following error types can be expected to be returned:
+//   - [applicationerrors.ApplicationIDNotValid] is returned if the application
+//     ID is not valid.
+//   - [applicationerrors.CharmOriginNotValid] is returned if the charm origin
+//     ID is not valid.
+//   - [applicationerrors.ResourceNameNotValid] if no resource name is provided in
+//     the args or if it doesn't match with a charm declared resource.
+//   - [applicationerrors.ResourceNotFound] if no resource with name exists for
+//     given application.
+func (s *Service) AddResources(
```

Comment:
issue: This functionality should be part of the CreateApplication method so it can be done in a single transaction. Helps to avoid the PendingResource problem. 

We will need a new method for the charm revision updater which combines a charm and resource data - should a different ticket.

Reply:
Code deleted.